app/src/main/java/com/ecosystem/kin/app/MainActivity.java

```diff
@@ -174,25 +174,37 @@ private void copyToClipboard(CharSequence textToCopy) {
 
     private void getBalance() {
         try {
+            //Get Cached Balance
+            int cachedBalance = Kin.getCachedBalance();
+            setBalanceWithAmount(cachedBalance);
+
             Kin.getBalance(new Callback<Integer>() {
                 @Override
                 public void onResponse(Integer balance) {
                     enableView(balanceView, true);
-                    balanceView.setText(getString(R.string.get_balance_d, balance));
+                    setBalanceWithAmount(balance);
                 }
 
                 @Override
                 public void onFailure(Throwable t) {
                     enableView(balanceView, true);
-                    balanceView.setText(R.string.failed_to_get_balance);
+                    setBalanceFailed();
                 }
             });
         } catch (TaskFailedException e) {
-            balanceView.setText(R.string.failed_to_get_balance);
+            setBalanceFailed();
             e.printStackTrace();
         }
     }
 
+    private void setBalanceFailed() {
+        balanceView.setText(R.string.failed_to_get_balance);
+    }
+
+    private void setBalanceWithAmount(int cachedBalance) {
```

Comment:
setBalanceWithAmount(int balance) it will display cache or fresh balance

Reply:
it just a helper method to set the text on the balance button.
it can be either the cached balance and the refreshed one.